GG/CMakeLists.txt

```diff
@@ -122,6 +122,7 @@ if(NOT BUILD_HEADLESS)
             $<$<VERSION_LESS:${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION},1.68>:GIGI_CONFIG_USE_OLD_IMPLEMENTATION_OF_GIL_PNG_IO>
             $<$<VERSION_GREATER:${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION},1.67>:BOOST_GIL_IO_ENABLE_GRAY_ALPHA>
             $<$<VERSION_GREATER:${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION},1.71>:BOOST_GIL_USES_MP11>
+            $<$<VERSION_GREATER:${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION},1.73>:BOOST_GIL_USES_VARIANT2>
```

Comment:
doesn't work for MSVC...

Reply:
MSVC should use boost from SDK

Reply:
SDK currently has 1.70 (or at least my version does), but it should be compatible with future Boost version updates.

Reply:
I suppose Freeorion MSVC project updates when switch to new SDK.

Reply:
probably, but I'd like to future-proof it now while we're looking at it.

Reply:
Maybe then we shouldn't use custom definitions based on boost version and use `BOOST_VERSION` in sources instead? Or define them somewhere in `config.h` file.

Reply:
I've removed definition from `CMakeLists.txt`.